devops/providers/azure-devops/templates/infrastructure/tasks/tf-ws-compute.yml

```diff
@@ -7,7 +22,7 @@ parameters:
 steps:
   - task: Bash@3
     name: ${{ parameters.stepName }}
-    displayName: Compute and output pipeline variable for Terraform Workspace
+    displayName: Workspace Name
     env:
       TERRAFORM_WORKSPACE_PREFIX: ${{ parameters.terraformWorkspacePrefix }}
       ENVIRONMENT_NAME: ${{ parameters.environmentName }}
```

Comment:
There is now a built-in pipeline system variable that should have the PR number. It should no longer be necessary to extract the PR number from the Build SourceBranch.

https://docs.microsoft.com/en-us/azure/devops/pipelines/build/variables?view=azure-devops&tabs=yaml#system-variables

Reply:
(the PR Number system var was broken when we originally wrote this code, the SourceBranch extraction was intended to be a workaround)

Reply:
Giving `System.PullRequest.PullRequestNumber` a try.

Reply:
This is confirmed to work! great callout. Note: I've gone ahead and disabled PR isolation, otherwise our PRs take a super long time for the isolated template. But the field is still explicitly called out in the template as a toggle-able switch